### DISCUSSION TITLE
High latency on puma

### DISCUSSION THREAD

**Discussion by MarwanTukhta** [puma/puma#3316] [Q&A]
**Describe the bug**
Puma seem to be adding latency to responses when we replaced passenger with puma, approximately 100ms more, everything else seem to be fine, even memory utilization is down compared to passenger, what do you think could be the reason for this?

**Puma config:**

```
max_threads_count = 1
min_threads_count = 1
threads min_threads_count, max_threads_count

port 3000

environment "production"

workers 30

on_worker_boot do
  SemanticLogger.reopen
# this made it worse
# ActiveRecord::Base.establish_connection if defined?(ActiveRecord)
end

fork_worker = true

preload_app!

activate_control_app

# this made it worse
# queue_requests false


bind 'unix:///var/run/puma.sock'
pidfile "/home/app/tmp/pids/puma.pid"
state_path "/home/app/tmp/pids/puma.state"

plugin :tmp_restart
plugin :yabeda
plugin :yabeda_prometheus
```

**To Reproduce**
above configuration on a rails app on a k8s pod with 
```
  cpu: 5250m
  memory: 13Gi
```

Run it with:

```
bundle exec puma -C  ./config/puma.rb
```

**Expected behavior**
have the same latency as passenger

**Desktop (please complete the following information):**
 - OS: Linux ubuntu 20.04
 - Puma Version 6.2.2

**Comment by dentarg** [comment]
Hi, you may find threads like https://github.com/rails/rails/issues/50450 and https://github.com/puma/puma/issues/2311 insightful (any maybe other issues here on the puma repo, 

However, without sharing more info (ruby version? passenger setup?) and most importantly an example app (and benchmarks? there's [one](https://github.com/speedshop/threadbench) linked in the rails issue) that others can run I'm not sure anyone can help you finding the culprit for these 100ms

**Comment by MarwanTukhta** [comment]
Thanks @dentarg, these are the versions for each tool
Ruby: 2.7.7
RVM
Rails 6.0.6.1
Phusion Passenger(R) 6.0.17 with nginx with 25 workers.

**Comment by dentarg** [comment]
@MarwanTukhta I converted the issue to a discussion because it is not an actionable issue for Puma. You will need to investigate the problem on your own. It is not possible for anyone else to help out unless they have access to the exact same environment as you have. 

Your setup having less latency with Passenger than Puma might not be a problem with Puma itself.
